app/components/InputStyler/ArrayFormatter.tsx

```diff
@@ -146,52 +162,139 @@ export default function ArrayFormatter(
     setRawData(newData);
   };
 
+  // Utility function to remove an item from an array
+  const removeItemFromArray = (arrayPath: string) => {
+    const currentArray = rawData ? getValueFromPath(arrayPath, rawData) || [] : [];
+    currentArray.pop();
+    const newData = { ...rawData, [arrayPath]: currentArray };
+    setRawData(newData);
+  };
+
   if (editable) {
     if (Array.isArray(parsedValue)) {
```

Comment:
This is an outdated critique since this is not part of this PR, but it tends to cleaner to use guard clauses for this sort of thing. Like https://medium.com/nirman-tech-blog/using-guard-clauses-for-cleaner-react-code-77360e648659

Reply:
Cool! Refactored and will keep in mind going forward.